app/src/main/java/net/gsantner/opoc/format/todotxt/SttCommander.java

```diff
@@ -412,5 +417,15 @@ private int compare(List<String> x, List<String> y) {
             return x.get(0).compareTo(y.get(0));
 
         }
+        private int compare(String x, String y) {
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+            try {
+                Date xAsDate = sdf.parse(x);
```

Comment:
any of both may be null or a invalid format. In that case it will be either nullpointer or invalidformatexception.

You could do e.g. two try catches, and assign 0 to x/y in case of catch clause.

Reply:
Or even better, simple string comparision should be enough!

As the format is yyyy - mm - dd, its already in good shape for default (aphabetical) sort. The format prefixes also 0 e.g. `07`, so ideal.

Reply:
Date format in todo.txt is part of specification and not to be changed But it's fine